doc/release-notes.md

```diff
@@ -0,0 +1,388 @@
+policy sometimes called "first-seen".
+
+The Bitcoin Protocol does not, and cannot, provide any assurance that
+the first version of an unconfirmed transaction seen by a particular
+node will be the version that gets confirmed.  If there are multiple
+versions of the same unconfirmed transaction available, only the miner
+who includes one of those transactions in a block gets to decide which
+version of the transaction gets confirmed.
+
+Despite this lack of assurance, multiple merchants and services today
+still make this assumption.
+
+There are several benefits to users from removing this *first-seen*
+simplification.  One key benefit, the ability for the sender of a
+transaction to replace it with an alternative version paying higher
+fees, was realized in [Bitcoin Core 0.12.0][] (February 2016) with the
+introduction of [BIP125][] opt-in Replace By Fee (RBF).
+
+Since then, there has been discussion about completely removing the
+first-seen simplification and allowing users to replace any of their
+older unconfirmed transactions with newer transactions, a feature called
+*full-RBF*.  This release includes a `mempoolfullrbf` configuration
+option that allows enabling full-RBF, although it defaults to off
+(allowing only opt-in RBF).
+
+Several alternative node implementations have already enabled full-RBF by
+default for years, and several contributors to Bitcoin Core are
+advocating for enabling full-RBF by default in a future version of
+Bitcoin Core.
+
+As more nodes that participate in relay and mining begin enabling
+full-RBF, replacement of unconfirmed transactions by ones offering higher
+fees may rapidly become more reliable.
+
+Members of this project strongly recommend that merchants and services
+not accept unconfirmed transactions as final, and if they insist on doing so,
+to take the appropriate steps to ensure they have some recourse or plan for
+when their assumptions do not hold.
+
+[Bitcoin Core 0.12.0]: https://bitcoincore.org/en/releases/0.12.0/#opt-in-replace-by-fee-transactions
+[bip125]: https://github.com/bitcoin/bips/blob/master/bip-0125.mediawiki
+
+Notable changes
+===============
+
+P2P and network changes
+-----------------------
+
+- To address a potential denial-of-service, the logic to download headers from peers
+  has been reworked.  This is particularly relevant for nodes starting up for the
+  first time (or for nodes which are starting up after being offline for a long time).
+
+  Whenever headers are received from a peer that have a total chainwork that is either
+  less than the node's `-minimumchainwork` value or is sufficiently below the work at
+  the node's tip, a "presync" phase will begin, in which the node will download the
+  peer's headers and verify the cumulative work on the peer's chain, prior to storing
+  those headers permanently. Once that cumulative work is verified to be sufficiently high,
+  the headers will be redownloaded from that peer and fully validated and stored.
```

Comment:
> redownloaded

Just to forestall concern about IBD (or more generally, sync) taking much longer, would it be worth mentioning that header sync takes a small fraction of the total sync time? (In my experience, it's just a few minutes out of many hours or  even days.)

Reply:
Going to leave this as is